pygmt/tests/test_solar.py

```diff
@@ -0,0 +1,101 @@
+"""
+Tests for solar.
+"""
+import datetime
+
+import pytest
+from pygmt import Figure
+from pygmt.exceptions import GMTInvalidInput
+
+
+@pytest.mark.mpl_image_compare
```

Comment:
The terminator lines are very complicated. How can we verify that these baseline images are correct?

Reply:
Isn't this the issue in general with comparing to baseline images? What's the work around to make the terminator more straightforward?

Reply:
Yes, it's a general issue for all plotting tests. For solar, I think we could have the following two tests:

1. `test_solar_ternimators`: plot the 4 terminators using different line styles. I feel it's much easier to visually check the correctness of the image
2. `test_solar_terminator_datetime`: check that different datetime types (string, datetime et al) produce the same figure.

For test 2, we can use [`pytest.mark.parametrize`](https://docs.pytest.org/en/stable/parametrize.html) to simplify the test, and use the `filename` parameter of the `mpl_image_compare` decorator. For example (I don't run it, so not sure if it really works),
```python
@pytest.mark.mpl_image_compare(filename="test_solar_terminator_datetime.png")
@pytest.mark.parametrize(
    "terminator_datetime", 
    ["1990-02-17 04:25:00", 
     datetime.datetime(year=1990, month=2, day=17, hour=4, minute=25, second=0)
    ]
)
def test_solar_set_terminator_datetime(terminator_datetime):
    ...
```
The test2 only produces one image, and it also checks that different datetime types produce the same results.

Reply:
updated in 886c15f